src/views/Add.vue

```diff
@@ -0,0 +1,176 @@
+            </v-btn>
+
+            <v-btn color="warning" class="mt-4" block to="/resources">
+              Go Back
+            </v-btn>
+          </div>
+        </v-form>
+      </v-sheet>
+    </v-card>
+  </v-container>
+</template>
+
+<script>
+import { ref, computed, watch, inject } from "vue";
+import Pearl from "../components/AddResources/Pearl.vue";
+import PreciousMetal from "../components/AddResources/PreciousMetal.vue";
+import Gemstone from "../components/AddResources/Gemstone.vue";
+import LinkingPart from "../components/AddResources/LinkingPart.vue";
+import { useRoute } from "vue-router";
+import { useStore } from "vuex";
+
+export default {
+  components: {
+    Pearl,
+    PreciousMetal,
+    Gemstone,
+    LinkingPart
+  },
+  props: {
+    id: String,
+  },
+  setup(props) {
+    const store = useStore();
+    const route = useRoute();
+    const options = ref(["Pearl", "Gemstone", "PreciousMetal", "LinkingPart"]);
+    const pageTitle = ref(route.meta.title);
+    const getText = computed(() => (text1, text2) => {
+      return pageTitle.value === "Edit resource" ? text1 : text2;
+    });
+
+    const showSnackbar = inject("showSnackbar");
+    const selected = ref("");
+    const form = ref(null);
+    const sentData = ref(false);
+    const formData = computed(() => store.state.resources.resourceDetails);
+
+    if (props.id) {
+      const data = computed(() =>store.getters["resources/getResourceById"](props.id));
+      store.dispatch("resources/setResourceDetails", data.value);
+      const resourceDetail = computed(() => store.getters["resources/getResourceDetails"]);
+      selected.value = resourceDetail.value.clazz;
+    }
+
+    const handleSelectedOption = (newValue) => {
+      selected.value = newValue;
+    };
+
+    watch(selected, (newValue) => {
+      //Reset input data on selection change
+      store.dispatch("resources/updateFormData", {});
+    });
+
+    const resData = () => {
+      if (form.value) {
+        console.log(form.value);
+        form.value.reset();
+        form.value.resetValidation();
+      }
+    };
+
+    const handleSubmit = async () => {
+      const { valid } = await form.value.validate();
+      if (valid && pageTitle.value === "Add resource") {
+        store.dispatch("resources/setResourceDetails", {
+          clazz: selected.value,
+          ...formData.value,
+        });
+        try {
+          await store.dispatch("resources/AddResources", {
+            clazz: selected.value,
+            ...formData.value,
+          });
+          infoSent();
+          resData();
+        } catch (error) {
+          showSnackbar({
+            message: "Failed to send resource.",
+            color: "error",
+            timeout: 4000,
+            location: "top right",
+          });
+        }
+      } else if (valid && pageTitle.value == "Edit resource") {
+        const resourceDetail = computed(
+          () => store.getters["resources/getResourceDetails"]
```

Comment:
This resourceDetail is the same as formData fetched on line 86 (const formData = computed(() => store.state.resources.resourceDetails);)